MiniZincIDE/highlighter.h

```diff
@@ -22,6 +22,123 @@ struct Bracket {
     int pos;
 };
 
+class ThemeColor{
+public:
+    QColor light;
+    QColor dark;
+    ThemeColor(){
+        light = Qt::red;
+        dark = Qt::red;
+    }
+
+    ThemeColor(QColor lightAndDark){
+        light = lightAndDark;
+        dark = lightAndDark;
+    }
+
+    ThemeColor(QColor l, QColor d){
+        light = l;
+        dark = d;
+    }
+    QColor get(bool darkMode){
+        return darkMode?dark:light;
+    }
+};
+
+struct Theme
+{
+        ThemeColor textColor, // Color of normal text
+        keywordColor, // Color of MiniZinc keywords such as "in" "where" "constraints"
+        functionColor, // Color of words that are followed by a function call, such as "forall()"
+        stringColor, // Color of strings
+        commentColor, // Color of comments %
+        backgroundColor, // Color of background
+        errorColor, //Color of error messages
+        lineHighlightColor,  // Color of highlighted line
+        foregroundActiveColor, // color of active line number
+        foregroundInactiveColor, // color of inactive line number
+        textHighlightColor,      // color of highlighted text
+        bracketsMatchColor, // Color of matching brackets
+        bracketsNoMatchColor //Colors of dangling brackets
+        ;
+        Theme(): Theme(Qt::white,Qt::darkGreen, Qt::blue, Qt::darkRed, Qt::red, Qt::white, Qt::lightGray) {}
+
+        Theme (QColor _textColor,
+               QColor _keywordColor,
+               QColor _functionColor,
+               QColor _stringColor,
+               QColor _commentColor,
+               QColor _backgroundColor,
+               QColor _lineHighlightColor):
+            Theme(ThemeColor(_textColor,_textColor),
+            ThemeColor(_keywordColor,_keywordColor),
+            ThemeColor(_functionColor,_functionColor),
+            ThemeColor( _stringColor,_stringColor),
+            ThemeColor(_commentColor,_commentColor),
+            ThemeColor(_backgroundColor,_backgroundColor),
+            ThemeColor(_lineHighlightColor,_lineHighlightColor),
+            ThemeColor(Qt::red,Qt::red))
```

Comment:
Any reason this is not specifyable via a QColor?

Reply:
A Color is just a single color. The point of `ThemeColor` is to create a pair of colors, one for light and one for dark mode. Maybe this could have been done by extending the QColor class, but that seems like unnecessary work :)

Reply:
I meant that this constructor can be used to set the theme color using a single QColor per color class, except for error color which is hard-coded.

Reply:
Ah yes sorry. 

Yeah I guess this is a bit strange. One can specify the error color using the contractor below, I just found that for my cases the error color is always red, so I made this constructor where this was just hard-coded so that I did not have to specify it. If this is too weird, we can just remove this constructor and use the one below.

By the way, going over the code now I see that there is actually no constructor for specifying all colors.
The colors `bracketsMatchColor`, `bracketsNoMatchColor`, `textHighlightColor`, `foregroundActiveColor`, and `foregroundInactiveColor` are derived from the other colors but the formulas are actually not safe for all input (safe in the sense that the colors can end up having bad contrasts). But adding the full constructor is trivial if ever required.